lib/setupHooks/installCargoArtifactsHook.sh

```diff
@@ -1,20 +1,46 @@
 compressAndInstallCargoArtifactsDir() {
   local dir="${1:?destination directory not defined}"
   local cargoTargetDir="${2:?cargoTargetDir not defined}"
+  local prevArtifacts="${3}"
 
   mkdir -p "${dir}"
 
   local dest="${dir}/target.tar.zst"
   echo "compressing ${cargoTargetDir} to ${dest}"
   (
     export SOURCE_DATE_EPOCH=1
-    tar --sort=name \
+
+    dynTar() {
+      if [ -n "${doCompressAndInstallFullArchive}" ]; then
+        >&2 echo "compressing and installing full archive as requested"
+        tar "$@" "${cargoTargetDir}"
+      elif [ "$(uname -s)" == "Darwin" ]; then
+        # https://github.com/rust-lang/rust/issues/115982
+        >&2 echo "incremental zstd compression not currently supported on Darwin: https://github.com/rust-lang/rust/issues/115982"
+        tar "$@" "${cargoTargetDir}"
+      elif [ -z "${prevArtifacts}" ]; then
+        >&2 echo "no previous artifacts found, compressing and installing full archive"
+        tar "$@" "${cargoTargetDir}"
+      else
+        >&2 echo "linking previous artifacts ${prevArtifacts}"
+        ln -s "${prevArtifacts}" "${dest}.prev"
```

Comment:
This (currently) links to the directory containing `target.tar.zst`, not the file itself:

```
fedimint-workspace-build> linking previous artifacts /nix/store/bxlki6y5zn5lgj2dlcxq2jh687qxzxzw-fedimint-deps-0.0.1
```

which will make the `if [ -f $candidate ]; ...` ignore it.

Reply:
We were missing the logic for handling other kinds of `.prev` artifacts (including other directories). I added some new logic in if you want to try it out (still need to incorporate the no-rebuild logic in the tests to validate this won't regress)

Reply:
Simplified this drastically, there's no need to be so hyper generic (it makes reasoning about the inputs really difficult). We can always revisit if its necessary to be more generic

Based on the logs this should be working now!